### ISSUE TITLE
FBP reconstructed tensor no grad

### ISSUE THREAD

**Issue by sunly92** [LLNL/LEAP#121]
In this example https://github.com/LLNL/LEAP/blob/main/demo_leaptorch/test_project_and_FBP.py, the FBP-reconstructed  tensor f_recon is not differentiable:  
```
    # run filtered back projection
    f_recon = proj.fbp(g_th)
    ......
```

Output of f_recon.requires_grad is False. I am wondering if I can simply solve this by calling f_recon.requires_grad_() or the fbp implemented in LEAP is no differentiable as it is.

**Comment by kylechampley** [comment]
The FBP function in the Projector class is not differentiable, it is just a feature that you can use.  If you want the differentiable version, then there is a class called FBP in leaptorch.py- use this instead.

**Comment by sunly92** [comment]
> The FBP function in the Projector class is not differentiable, it is just a feature that you can use. If you want the differentiable version, then there is a class called FBP in leaptorch.py- use this instead.

I checked the gradient of the FBP-reconstructed tensor using class FBP in leaptorch.py as you said, and it works just well. But I encountered some problems:
I am trying to implement a so-called FBP layer in a Pytorch pipeline: an intermediate CT image (maybe reconstructed from sparse-view or limited-view sinogram) is fed into this layer, and then it is projected onto the sinogram domain. The true CT measures on certain angles simply replace the corresponding ones and a filtered back projection will be performed to get a "corrected" CT image. 
In this case, Do I have to define two classes: "Projector" and "FBP" in order to make this layer fully differentiable? The same CT geometry and volume settings should be passed twice to them, which is kind of confusing to me. Please kindly correct me if I am wrong. I really appreciate it if you could provide some help.

**Comment by kylechampley** [comment]
I'm not sure what the correct implementation should be, but all the building blocks are here for you to implement a custom differentiable layer.

**Comment by kylechampley** [comment]
I'm going to close this issue due to lack of activity, but feel free to reopen it if something comes up.
